### DISCUSSION TITLE
Output carbonstock at grid level

### DISCUSSION THREAD

**Discussion by KeLiu7** [magpiemodel/magpie#452] [Help]
Dear MAgPIE community,

I'm trying to extract carbon stock using function `carbonstock` at grid level.  However, the level of aggregation in this function only includes "cell", "reg", "glo" and "regglo". Is there any other way to disaggregate it at grid level?

I tried with `gdxAggregate()` but it does not work...

Many thanks for your suggestions!

**Comment by bodirsky** [comment]
gdxAggregate should work with any resolution, you just need to think about the right weight... what exactly did you try?

If you want to do it properly, I think the best way would be to read in the carbon densities on 0.5° and use those as disaggregation weight. 
If you have the input data tgz  on 0.5°, you can pull out the file you need like this

tgz <- paste0("/p/projects/landuse/data/input/archive/",strsplit(cfg$input[1],split = "c200")[[1]][[1]],"0.5.tgz")
print(paste0("trying to extract lpj_yields_0.5.mz from ",tgz))
untar(tarfile = tgz, files = "lpj_yields_0.5.mz", exdir=outputdir)

(instead of yields, you shoudl take the carbon densities of the different land types)

**Reply by KeLiu7** [reply]
Thanks Bodirsky. I just tried again with gdxAggregate and now it works. Could you please help check if the codes below are right for extracting Carbon stock in vegetation soil? Thanks in advance.

vegc_carbon <- carbonstock(gdx = "fulldata.gdx", sum_cpool=FALSE)[,,"vegc"]

grid_vegc<- gdxAggregate(gdx="fulldata.gdx",x = vegc_carbon, to="grid",weight="land", absolute=TRUE,dir=".")
